pkg/controller/registry/registry_controller_suite_test.go

```diff
@@ -62,21 +65,38 @@ func TestMain(m *testing.M) {
 
 // SetupTestReconcile returns a reconcile.Reconcile implementation that delegates to inner and
 // writes the request to requests after Reconcile is finished.
-func SetupTestReconcile(inner reconcile.Reconciler) (reconcile.Reconciler, chan reconcile.Request) {
+func SetupTestReconciler(inner reconcile.Reconciler) (reconcile.Reconciler, chan reconcile.Request) {
+
 	requests := make(chan reconcile.Request)
-	fn := reconcile.Func(func(req reconcile.Request) (reconcile.Result, error) {
+	recFn := reconcile.Func(func(req reconcile.Request) (reconcile.Result, error) {
 		result, err := inner.Reconcile(req)
 		requests <- req
 		return result, err
 	})
-	return fn, requests
+
+	return recFn, requests
+}
+
+func  SetupTestManager(t *testing.T) manager.Manager {
+
+	// Setup the Manager and Controller.  
+	mgr, err := manager.New(cfg, manager.Options{})
+
+	if err != nil {
+		t.Errorf("Error creating manager %v", err)
+	}
+
+	return  mgr
 }
 
 // StartTestManager adds recFn
-func StartTestManager(mgr manager.Manager, g *gomega.GomegaWithT) chan struct{} {
+func StartTestManager(t *testing.T, mgr manager.Manager) chan struct{} {
 	stop := make(chan struct{})
 	go func() {
-		g.Expect(mgr.Start(stop)).NotTo(gomega.HaveOccurred())
+		err := mgr.Start(stop)
+		if err != nil {
+			t.Errorf("Failed to start Manager: %v", err)
```

Comment:
Only for reporting. I haven't found how to make the test fail in this situation which should be the best option, because without the manager, the test won't run.

Reply:
To clarify. The call to `mgr.Start(stop)` will return an error only if there's an error, otherwise, it will block until the `stop` channel is closed. Therefore, trying to return the error in a channel:
```
func StartTestManager(t *testing.T, mgr manager.Manager) (chan struct{}, chan error) {
        stop := make(chan struct{})                                                       
        errCh := make(chan error)                                                         
        go func() { 
                err := mgr.Start(stop)
                errCh <- err                                                              
        }()                                                                               
        return stop, errCh                                                                
}                                                                                         
```
Won't work because in  `TestReconcile` the following code blocks:
```
stop, errorCh := StartTestManager(t, mgr)
err = <- erroCh
if err != nil {
     t.Fatalf("Error starting manager %v", err)
}
defer close(stop)
```

Reply:
ok lets' resolve this ths @pablochacin